README.md

```diff
@@ -2,42 +2,339 @@
+
+`JustRelax.jl` is not yet registered (we are in the process), however it can be installed by cloning the repository. If you have navigated the terminal to the directory you cloned `JustRelax.jl`, you can test the package by running the following commands:
+
+```julia
+using JustRelax
+julia> ] 
+  pkg> activate .
+  pkg> instantiate
+  pkg> test JustRelax
+```
+The test will take a while, so grab a coffee or tea. 
+
+## Usage
+
+The package relies on [JustPIC.jl](https://github.com/JuliaGeodynamics/JustPIC.jl) for the particle in cell method. Depending on where you want to run `JustRelax.jl` (CPU or GPU) you will need to specify the Backend. Run this command in the REPL and restart Julia:
+
+```julia
+  set_backend("Threads_Float64_2D")       #running on the CPU
+  set_backend("CUDA_Float64_2D")          #running on an NVIDIA GPU
+  set_backend("AMDGPU_Float64_2D")        #running on an AMD GPU
+```
+After you have run your desired line and restarted Julia, there should be a file called `LocalPreferences.toml` in the directory together with your `Project.toml` and `Manifest.toml`. This file contains the information about the backend you want to use. If you want to change the backend, you can run the command again. 
+
+
+As stated above, the parallelisation is done using ParallelStencil.jl. Therefore, `JustRelax.jl` has as an environment already setup with which you can specify the dimension of your problem (2D or 3D) and the backend (CPU or GPU). The following commands are available:
+
+```julia
+  model = PS_Setup(:Threads, Float64, 2)  #running on the CPU in 2D
+  environment!(model)
+
+  model = PS_Setup(:CUDA, Float64, 2)     #running on an NVIDIA GPU in 2D
+  environment!(model)
+
+  model = PS_Setup(:AMDGPU, Float64, 2)   #running on an AMD GPU in 2D
+  environment!(model)
+```
+
+## Shear Band 2D example
+
+This example benchmark test displays how the package can be used to simulate shear bands. The example is based on the [ShearBands2D.jl](miniapps/benchmarks/stokes2D/shear_band/ShearBand2D.jl). 
+
+The `CUDA.allowscalar(false)` is needed to avoid the use of scalar values in the code which would run slowly on a CPU and is disallowed on GPU systems such as HPC clusters. 
+
+```julia
+using CUDA
+CUDA.allowscalar(false)
+
+using GeoParams, GLMakie, CellArrays
+using JustRelax, JustRelax.DataIO
+```
+Setup of environment and general solution for the benchmark test.
+
+```julia
+# setup ParallelStencil.jl environment
+model  = PS_Setup(:CUDA, Float64, 2)
+environment!(model)
+
+# HELPER FUNCTIONS ---------------------------------------------------------------
+solution(ε, t, G, η) = 2 * ε * η * (1 - exp(-G * t / η))
+
+```
+The function `init_phases!` initializes the phases within cell arrays. The function is parallelized with `@parallel_indices (i,j) ` and `@parallel (JustRelax.@idx ni)`. In this case, this is the only function that needs to be taylored to the specific problem, everything else is handled by `JustRelax.jl` itself. 
```

Comment:
"...is parallelized with the `@parallel_indices`  macro from ParallelStencil.jl."